### PR TITLE
Continue adding logging

### PR DESCRIPTION
**Pull request by kassianh**
Logging is done for any action or decision.

- Imperative statements are used for logging actions, simple statements for decisions
- Logging is done immediately before the action occurs
- Logging is done AFTER the decision has been made
	- Eg: In an if statement make log directly after entering saying what has happened
	- Simple statement - non-imperative. Eg: 'Changed' flag was set
- Not all fields are required
- Format is: 
	- level - trace, debug
	- time - Unix or RC3339
	- stack (optional) - Path of stack
	- action - Name of function where the action is found. With `()` to improve readibility
	- fileContext (optional) - Absolute path of file if reading a file
	- configFile (optional) - Config file associated with action
	- configDir (optional) - Config directory if relevant
	- workingDirectory (optional) - Working directory - especially used if stack not relevant
	- path (optional) - Path/directory if relevant
	- command (optional) - Command if relevant
	- reference (optional) - Reference if relevant
	- id (optional) - ID if relevant
	- attribute (optional)
	- message - Imperative/simple statement describing what is about to happen/has happened.. Period `.` at end to improve readibility.
- Things not to log:
	- return statements
	- errrors (are handled differently)
- Msg should not include variables - all msgf variables should be in a context field
Other relevant info should also be logged

### CODE REVIEW

**Review by katcipis — COMMENTED** [review]
Gave a few suggestions on naming/avoiding duplication. Overall LGTM since we can always improve later and it doesn't break anything. Will leave approval to @mariux so he can double check if it is enough as a starting point to help him.

**Comment by mariux** [comment]
@kassianh lgtm... please rebase... ;) sorry i destroyed some ;) please check what changed

**Comment by mariux** [comment]
@kassianh you can just globally accept all current changes when resolving.... i guess tests need to be adjusted after a rebase to main...

**Review by mariux — CHANGES_REQUESTED** [review]
pls rebase and fix tests afterwards

**Comment by katcipis** [comment]
> @kassianh you can just globally accept all current changes when resolving.... i guess tests need to be adjusted after a rebase to main...

After we merge this: https://github.com/mineiros-io/terramate/pull/147
Adding logs should not affect tests anymore, so hopefully no need for test adjusting :-)

File: exported_locals.go [hunk 1]
```diff
@@ -67,10 +79,27 @@ func (e ExportedLocalValues) Attributes() map[string]cty.Value {
...
+		Str("path", rootdir).
+		Logger()
+
+	logger.Trace().
+		Msg("Get config file path.")
 	cfgpath := filepath.Join(rootdir, cfgdir, config.Filename)
+
+	logger.Debug().
+		Str("configFile", cfgpath).
+		Msg("Parse export as locals blocks.")
```
[the remark this answers is not in the document]
Comment (kassianh) [reply exported_locals.go:93]: Good idea, thanks.

**Review by mariux — APPROVED** [review]
lgtm

**Pull request merged**